collectors/PropertiesCollector.py

```diff
@@ -0,0 +1,95 @@
+from BaseCollector import BaseCollector
+import logging
+
+logger = logging.getLogger('vrops-exporter')
+
+
+class PropertiesCollector(BaseCollector):
+
+    def __init__(self):
+        super().__init__()
+
+    def get_uuids(self):
+        raise NotImplementedError("Please Implement this method")
+
+    def set_labels(self, resource_id: str, project_ids: list):
+        raise NotImplementedError("Please Implement this method")
+
+    def collect(self):
+        logger.info(f'{self.name} starts with collecting the metrics')
+
+        token = self.get_target_tokens()
+        token = token.setdefault(self.target, '')
+        if not token:
+            logger.warning(f'skipping {self.target} in {self.name}, no token')
+            return
+
+        uuids = self.get_uuids()
+        metrics = self.generate_metrics(label_names=self.label_names)
+        project_ids = self.get_project_ids_by_target() if self.project_ids else []
+        values, api_responding = self.vrops.get_latest_properties_multiple(self.target,
+                                                                           token,
+                                                                           uuids,
+                                                                           [m for m in metrics],
+                                                                           self.name)
+        yield self.create_api_response_metric(self.name, api_responding)
+
+        if not values:
+            logger.warning(f'No values in the response for {self.name}. API code: {api_responding}')
+            return
+
+        no_match_in_config = set()
+        values_received = set()
+
+        for resource in values:
+            resource_id = resource.get('resourceId')
+
+            for value_entry in resource.get('property-contents', {}).get('property-content', []):
+                labels = self.set_labels(resource_id, project_ids)
+
+                statkey = value_entry.get('statKey')
+                values_received.add(statkey)
+
+                metric_data = value_entry.get('data', [False])[0]
+                metric_value = value_entry.get('values', [False])[0]
+
+                if statkey in metrics:
+                    # enum metrics
+                    if metrics[statkey]['expected']:
+                        relabel = metrics[statkey]['gauge']._labelnames
+                        if 'state' not in relabel:
+                            relabel += ('state',)
```

Comment:
A little bit unlucky, that we have to do it here, but it is the tradeoff to have it in a central place.
Is the `,` here really needed `('state',)` ?

Reply:
Yes, it is a tuple object. And when you add one entry to a tuple, this is how you do it.